core/validation/initial_check.py

```diff
@@ -0,0 +1,150 @@
+    if sheet_a2 is not None:
+        details_dict["Are you filling this in for a Town Deal or Future High Street Fund?"] = sheet_a2.iloc[5][4]
+        details_dict["Please select your place name"] = sheet_a2.iloc[6][4]
+        details_dict["A2"] = "valid"
+    else:
+        details_dict["Are you filling this in for a Town Deal or Future High Street Fund?"] = None
+        details_dict["Please select your place name"] = None
+        details_dict["A2"] = None
+
+    return details_dict
+
+
+def pre_transformation_check(submission_details: dict[str, str]) -> list[vf.ValidationFailure]:
+    """
+    Perform pre-transformation checks on the given workbook.
+
+    :param submission_details: A dictionary containing submission details.
+    :return: A list of validation failures encountered during pre-transformation checks.
+    """
+    pre_transformation_failures = []
+
+    check_form_type(submission_details, pre_transformation_failures)
+    check_round(submission_details, pre_transformation_failures)
+    check_form_input(
+        submission_details,
+        "Are you filling this in for a Town Deal or Future High Street Fund?",
+        pre_transformation_failures,
+    )
+    check_form_input(submission_details, "Please select your place name", pre_transformation_failures)
+    check_empty_sheet(submission_details, "A1", pre_transformation_failures)
+    check_empty_sheet(submission_details, "A2", pre_transformation_failures)
+
+    return pre_transformation_failures
+
+
+def check_form_type(submission_details: dict[str, str], failures: list[vf.ValidationFailure]) -> None:
+    """
+    Check if the form type is correct.
+
+    :param submission_details: A dictionary containing submission details.
+    :param failures: A list to store validation failures.
+    :return: None
+    """
+    expected_form_type = "Town Deals and Future High Streets Fund Reporting Template (v3.0)"
+    actual_form_type = submission_details["Form Version"]
+
+    if actual_form_type != expected_form_type:
+        failures.append(
+            vf.WrongFormTypeFailure(
+                expected_form_type=expected_form_type,
+                actual_form_type=actual_form_type,
+            )
+        )
+
+
+def check_round(submission_details: dict[str, str], failures: list[vf.ValidationFailure]) -> None:
+    """
+    Check if the round value is correct.
+
+    :param submission_details: A dictionary containing submission details.
+    :param failures: A list to store validation failures.
+    :return: None
+    """
+    reporting_period = submission_details["Reporting Period"]
+
+    if reporting_period not in REPORTING_PERIOD_DICT:
+        failures.append(
+            vf.InvalidReportingPeriodFailure(
+                invalid_reporting_period=reporting_period,
+                valid_reporting_periods=REPORTING_PERIOD_DICT.keys(),
+            )
+        )
+
+        return
+
+    round_value = REPORTING_PERIOD_DICT[reporting_period]
+
+    if round_value != 3:
```

Comment:
my feeling is this bit should be refactored out into a separate function that is passed the expected round of "3" as a param, hard-coding it like this makes the function very specific